lsp/nls/src/requests/hover.rs

```diff
@@ -16,14 +17,24 @@ use crate::{
     server::Server,
 };
 
-#[derive(Debug)]
+#[derive(Debug, Default)]
 struct HoverData {
     values: Vec<RichTerm>,
     metadata: Vec<FieldMetadata>,
-    span: RawSpan,
+    span: Option<RawSpan>,
     ty: Option<Type>,
 }
 
+impl Combine for HoverData {
+    fn combine(mut left: Self, right: Self) -> Self {
+        left.values.extend_from_slice(&right.values);
+        left.metadata.extend_from_slice(&right.metadata);
```

Comment:
I believe `extend_from_slice` is cloning, which is not required here. Any reason to not use `append` instead?

Reply:
No reason, good catch.